src/Adapter/CartRule/Repository/CartRuleRepository.php

```diff
@@ -87,6 +90,162 @@ public function get(CartRuleId $cartRuleId): CartRule
         return $cartRule;
     }
 
+    /**
+     * @param CartRuleId $cartRuleId
+     * @param RestrictionRuleGroup[] $restrictionRuleGroups
+     *
+     * @return void
+     */
+    public function setProductRestrictions(CartRuleId $cartRuleId, array $restrictionRuleGroups): void
+    {
+        // first remove all product restrictions concerning provided cart rule
+        $this->removeProductRestrictions($cartRuleId);
+
+        foreach ($restrictionRuleGroups as $restrictionRuleGroup) {
+            $this->connection->createQueryBuilder()
+                ->insert($this->dbPrefix . 'cart_rule_product_rule_group')
+                ->values([
+                    'id_cart_rule' => $cartRuleId->getValue(),
+                    'quantity' => $restrictionRuleGroup->requiredQuantityInCart,
+                ])
+                ->execute()
+            ;
+
+            $productRuleGroupId = $this->connection->lastInsertId();
+
+            foreach ($restrictionRuleGroup->restrictionRules as $restrictionRule) {
+                $this->connection->createQueryBuilder()
+                    ->insert($this->dbPrefix . 'cart_rule_product_rule')
+                    ->values([
+                        'id_product_rule_group' => ':productRuleGroupId',
+                        'type' => ':type',
+                    ])
+                    ->setParameter('productRuleGroupId', $productRuleGroupId)
+                    ->setParameter('type', $restrictionRule->type)
+                    ->execute()
+                ;
+
+                $productRuleId = $this->connection->lastInsertId();
+                $productRuleValues = [];
+                $checkedIds = [];
+                foreach ($restrictionRule->ids as $id) {
```

Comment:
What does ID stand for? IS it productId, categoryId? Or can it be used for different entities?

Reply:
yep, it can be category,product,attribute,supplier or manufacturer. Depends on $restriction->type

Reply:
Can you please replace this public field with a getter instead (and same in other commands, although it can be done in a dedicated PR for others) This one is a bit too generic, so maybe something like:

```suggestion
                foreach ($restrictionRule->getEntityIds() as $entityId) {
```

Reply:
Or `getRestrictedEntityIds` or `getRestrictedIds` I'm not sure